pkg/sources/adapter/azureeventhubssource/adapter.go

```diff
@@ -188,49 +204,74 @@ func (a *adapter) Start(ctx context.Context) error {
 	ctx = pkgadapter.ContextWithMetricTag(ctx, a.mt)
 
 	// listen to each partition of the Event Hub
-	for _, partitionID := range runtimeInfo.PartitionIDs {
-		connCtx, cancel := context.WithTimeout(ctx, connTimeout)
-		_, err := a.ehClient.Receive(connCtx, partitionID, a.handleMessage,
-			eventhub.ReceiveWithLatestOffset(), eventhub.ReceiveWithConsumerGroup(a.ehConsumerGroup))
-		cancel()
-		if err != nil {
-			a.logger.Errorw("An error occurred while starting message receivers. "+
-				"Terminating all active receivers", zap.Error(err))
-
-			closeCtx, cancel := context.WithTimeout(ctx, drainTimeout)
-			defer cancel()
-			if err := a.ehClient.Close(closeCtx); err != nil {
-				a.logger.Errorw("An additional error occurred while terminating active "+
-					"Event Hub message receivers", zap.Error(err))
+	wg := sync.WaitGroup{}
+	a.logger.Info("Starting Event Hub message receivers for partitions ", runtimeInfo.PartitionIDs)
+	for _, partition := range a.runtimeInfo.PartitionIDs {
+		wg.Add(1)
+		go func(partitionID string) {
+			defer wg.Done()
+			err = a.processPartition(ctx, partitionID)
+			if err != nil {
+				a.logger.Errorf("Error processing partition %s: %v", partitionID, err)
 			}
-
-			return fmt.Errorf("starting message receiver for partition %s: %w", partitionID, err)
-		}
+		}(partition)
 	}
-
 	health.MarkReady()
+	wg.Wait()
+	return nil
+}
 
-	<-ctx.Done()
-	a.logger.Debug("Terminating all active Event Hub message receivers")
-
-	closeCtx, cancel := context.WithTimeout(context.Background(), drainTimeout)
-	defer cancel()
-	if err := a.ehClient.Close(closeCtx); err != nil {
-		return fmt.Errorf("terminating active Event Hub message receivers: %w", err)
+// processPartition processes events from a single partition of the Event Hub.
+func (a *adapter) processPartition(ctx context.Context, partitionID string) error {
+	partitionClient, err := a.ehClient.NewPartitionClient(partitionID, &azeventhubs.PartitionClientOptions{
+		StartPosition: azeventhubs.StartPosition{
+			Latest: to.Ptr(true),
+		},
+	})
+	if err != nil {
+		return err
 	}
+	defer func() {
+		if err := partitionClient.Close(ctx); err != nil {
+			a.logger.Errorf("closing partition client for partition %s: %v", partitionID, err)
+		}
+	}()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
```

Comment:
Exit on the context termination is a normal shutdown procedure, it should not return an error:

```suggestion
			// optional shutdown message
			// h.logger.Debug("Shutting down Azure Event Hubs adapter")
			return nil
```